programmers-coding-test/kimjiwon/알고리즘/거스름돈.js

```diff
@@ -0,0 +1,32 @@
+function giveChange(customerMoney, goodsPrice) {
+    const unitMoney = [10000 ,5000 ,1000, 500, 100, 50, 10];
+    const changeUnitMoney = new Map();
+
+    let sum = 0,
+    change = customerMoney - goodsPrice;
+
+    unitMoney.forEach(el => {
+        if (calcurate(change, el)) {
+            changeUnitMoney.set(`${el}`, calcurate(change, el));
+            change -= (el * calcurate(change, el)); 
+        }
+    });
+
+    [...changeUnitMoney].forEach(el => {
+        sum += el[1];
+        console.log(`${el[0]}원 : ${el[1]}개` );
+    });
```

Comment:
일반 포문으로 할 수 있다면, 굳이 forEach를 쓰지 않는 것이 가독성이 좋은 듯 하다.
혹시나 ForEach를 사용한 이유가 있는가? (궁금)
Sum보다는 사용한 동전의 개수를 카운팅하는 것이므로 count가 더 이해하기 쉬운 듯 하다 :)

```suggestion
   let cnt = 0;
   for (let unitMoney in changeUnitMoney) {
       cnt += changeUnitMoney[unitMoney];
       console.log(`${unitMoney}원 : ${changeUnitMoney[unitMoney]}`);
   }
```

Reply:
딕셔너리를 배열로 만들어야겠다는 고정관념땜시ㅎㅎㅎㅎ
그냥 써도 되는걸 이제 알아버렸어요